### PR TITLE
Add Spark 3.5.0 support

### PR DESCRIPTION
**Pull request by EnricoMi**
Waiting for Spark 3.5.0 release...

### CODE REVIEW

**Review by eejbyfeldt — APPROVED** [review]
Awsome!

**Comment by EnricoMi** [comment]
@WeichenXu123 PySpark 3.5.0 has been released, graphframes is ready to move to Spark 3.5.0.

**Comment by WeichenXu123** [comment]
@EnricoMi Thanks! I just released graphframe 0.8.3 for spark 3.4 yesterday. Would you help adding CI for spark 3.5 ? then I will release a new version for spark 3.5

**Comment by EnricoMi** [comment]
> Would you help adding CI for spark 3.5 ?

Sure, what else is needed other than this PR?

**Comment by WeichenXu123** [comment]
> > Would you help adding CI for spark 3.5 ?
> 
> Sure, what else is needed other than this PR?

Nothing else is needed. CI is for checking graphframe is compatible with spark 3.5.

**Review by WeichenXu123 — APPROVED** [review]

File: src/test/scala/org/graphframes/lib/ConnectedComponentsSuite.scala [hunk 1]
```diff
@@ -215,18 +215,31 @@ class ConnectedComponentsSuite extends SparkFunSuite with GraphFrameTestSparkCon
...
-    val cc = friends.connectedComponents
-    assert(cc.getIntermediateStorageLevel === StorageLevel.MEMORY_AND_DISK)
-
-    for (storageLevel <- Seq(StorageLevel.DISK_ONLY, StorageLevel.MEMORY_ONLY, StorageLevel.NONE)) {
-      // TODO: it is not trivial to confirm the actual storage level used
-      val components = cc
-        .setIntermediateStorageLevel(storageLevel)
-        .run()
-      assertComponents(components, expected)
+    // disabling adaptive query execution helps assertComponents
```
Comment (eejbyfeldt) [inline src/test/scala/org/graphframes/lib/ConnectedComponentsSuite.scala:218]: The need for this change here looks like a correctness bug in Spark 3.5.0. I created a PR with a attempted fix here: https://github.com/apache/spark/pull/43188

**Pull request merged**